### PR TITLE
[*] Custom Column Name - Fix charts and sorting with underscored true or field option

### PR DESCRIPTION
**Pull request by VincentMolinie**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request
- [ ] Write changes made in the CHANGELOG.md
- [ ] Create automatic tests
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by VincentMolinie** [comment]
[P3 - Snake Case DB - When using snaked_case in db forest-express-sequelize doesn't handle well columnName](https://trello.com/c/PPGCVuIG/3571-p3-snake-case-db-when-using-snakedcase-in-db-forest-express-sequelize-doesnt-handle-well-columnname)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # Change Log
 
 ## [Unreleased]
+### Fixed
+- Pie Chart - Fix groupBy/aggregate on a field with custom column name.
+- Time Chart - Fix groupBy/aggregate on a field with custom column name.
+- Leaderboard Chart - Fix groupBy on a field with custom column name.
+- List Records - Fix sort on a field with custom column name.
```
Comment (arnaudbesnier) [inline CHANGELOG.md]: I am not convinced that this is the exact fix description as the sorting on a "custom column name" already works properly.
Do you talk about the sorting on belongsTo fields having a "reference field" with a "custom column name"?

File: src/services/query-builder.js [hunk 2]
```diff
@@ -59,11 +63,17 @@ function QueryBuilder(model, opts, params) {
...
+        const [tableName] = schemaField.reference.split('.');
+        const associationSchema = Schemas.schemas[tableName];
+        const belongsToColumnName = Orm.getColumnName(associationSchema, fieldName);
+        return [[opts.sequelize.col(`${associationName}.${belongsToColumnName}`), order]];
       } else if (aliasName) {
-        return [[opts.sequelize.col(`${aliasName}.${params.sort}`), order]];
+        return [[opts.sequelize.col(`${aliasName}.${Orm.getColumnName(aliasSchema, params.sort)}`), order]];
       }
-      return [[params.sort, order]];
+      return [[Orm.getColumnName(schema, params.sort), order]];
```
Comment (VincentMolinie) [inline src/services/query-builder.js]: This fix the sort on list of record

File: src/services/query-builder.js [hunk 3]
```diff
@@ -59,11 +63,17 @@ function QueryBuilder(model, opts, params) {
...
+        const [associationName, fieldName] = params.sort.split('.');
+        const schemaField = (aliasSchema || schema).fields
+          .find(field => field.field === associationName);
+        const [tableName] = schemaField.reference.split('.');
+        const associationSchema = Schemas.schemas[tableName];
+        const belongsToColumnName = Orm.getColumnName(associationSchema, fieldName);
+        return [[opts.sequelize.col(`${associationName}.${belongsToColumnName}`), order]];
       } else if (aliasName) {
-        return [[opts.sequelize.col(`${aliasName}.${params.sort}`), order]];
+        return [[opts.sequelize.col(`${aliasName}.${Orm.getColumnName(aliasSchema, params.sort)}`), order]];
```
Comment (VincentMolinie) [inline src/services/query-builder.js]: This fix sort on related records

File: src/services/query-builder.js [hunk 4]
```diff
@@ -59,11 +63,17 @@ function QueryBuilder(model, opts, params) {
...
       if (params.sort.indexOf('.') !== -1) {
         // NOTICE: Sort on the belongsTo displayed field
-        return [[opts.sequelize.col(params.sort), order]];
+        const [associationName, fieldName] = params.sort.split('.');
+        const schemaField = (aliasSchema || schema).fields
+          .find(field => field.field === associationName);
+        const [tableName] = schemaField.reference.split('.');
+        const associationSchema = Schemas.schemas[tableName];
+        const belongsToColumnName = Orm.getColumnName(associationSchema, fieldName);
+        return [[opts.sequelize.col(`${associationName}.${belongsToColumnName}`), order]];
```
Comment (VincentMolinie) [inline src/services/query-builder.js]: This fix sort on belongsTo

File: CHANGELOG.md (hunk 1 above)
Comment (VincentMolinie) [reply CHANGELOG.md]: Yes it does fix it. I added some comment to the PR ;)

**Pull request merged**